processing/src/main/java/io/druid/query/timeboundary/TimeBoundaryQueryQueryToolChest.java

```diff
@@ -152,8 +137,10 @@ public boolean apply(T input)
       public byte[] computeCacheKey(TimeBoundaryQuery query)
       {
         final byte[] cacheKey = query.getCacheKey();
-        return ByteBuffer.allocate(1 + cacheKey.length)
+        final byte descending = query.isDescending() ? (byte)1 : 0;
+        return ByteBuffer.allocate(2 + cacheKey.length)
```

Comment:
@navis I don't think @nishantmonu51 made any comments about timeboundary, I think it was more a general comment. For queries where it doesn't make sense to have ordering we should probably not allow descending order. I doubt it makes sense to enable it for timeboundary.